flink-core/src/test/java/org/apache/flink/core/memory/ByteBufferUtilsTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ *  * Licensed to the Apache Software Foundation (ASF) under one
+ *  * or more contributor license agreements.  See the NOTICE file
+ *  * distributed with this work for additional information
+ *  * regarding copyright ownership.  The ASF licenses this file
+ *  * to you under the Apache License, Version 2.0 (the
+ *  * "License"); you may not use this file except in compliance
+ *  * with the License.  You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package org.apache.flink.core.memory;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+
+/**
+ * Tests for {@link ByteBufferUtils}.
+ */
+public class ByteBufferUtilsTest {
+
+	@Test
+	public void testBBWriteAndRead() {
+
+		ByteBuffer bb = ByteBuffer.allocateDirect(4096);
+		doTest(bb, 0);
+		doTest(bb, 1);
+		doTest(bb, 2);
+		doTest(bb, 3);
+		doTest(bb, 4);
+		doTest(bb, 5);
+		doTest(bb, 6);
+		doTest(bb, 7);
```

Comment:
One could have save a couple of lines if this were done in a loop. On the other hand, this makes things a bit more explicit.